### ISSUE TITLE
Creating a Custom Indicator: The Big 4

### ISSUE THREAD

**Issue by twopirllc** [twopirllc/pandas-ta#355]
In short, there are basically **four** steps to fully add an indicator to Pandas TA. For concrete examples, I recommend looking at [@rengel8's RSX](https://github.com/twopirllc/pandas-ta/pull/182/files) and [@whubsch's MCG](https://github.com/twopirllc/pandas-ta/pull/190/files) indicators. Essentially their indicator submission is some combination of steps outlined below.

# Creating a Custom Indicator: The Big 4

This is a step by step guide to creating a Custom Indicator to provide both Standard and _Pandas TA DataFrame_ Extensionable Indicators. So lets create a "hypothetical indicator" named ```ni(close, length=10, **kwargs)``` that belongs in the _Trend_ Category. It's implementation is up to you, however at a minimum it should only rely on Pandas's [dependencies](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html) (typically Pandas or NumPy calculations).

## **Step One:** Adding the Indicator to it's Category
For ```ni(close, length=None, **kwargs)``` to be evaluated the Standard way, we need to do two things:

1. Add the line ```from .ni import ni``` to [```pandas_ta/trend/__init__.py```](https://github.com/twopirllc/pandas-ta/blob/360a26d71c83fe87e4042e4f86298b1dc3023704/pandas_ta/trend/__init__.py#L12)
1. Create the file: ```pandas_ta/trend/ni.py```.
    1. Copy a similar indicator _as_ a template for the **ni** indicator specification and then modify it.

Now it should be callable the _Standard_ way like a TA Lib indicator:

```python
import pandas as pd
import pandas_ta as ta

df = # ohlcv data
nidf = ta.ni(df["close"]) #  = ta.ni(df["close"], 10)
```

## **Step Two:** Adding the Indicator to the Pandas TA DataFrame

The next two steps are **required** to link ```ni(close, length=10, **kwargs)``` with Pandas TA DataFrame. This allows you to call it from the **ta** extension or from the **ta.strategy()** extension method.

1. Add "ni" to the [Category dict](https://github.com/twopirllc/pandas-ta/blob/360a26d71c83fe87e4042e4f86298b1dc3023704/pandas_ta/__init__.py#L68) under the "trend" key in ```pandas_ta/__init__.py```
1. Add the [**ni** method](https://github.com/twopirllc/pandas-ta/blob/360a26d71c83fe87e4042e4f86298b1dc3023704/pandas_ta/core.py#L1401) to ```pandas_ta/core.py```.
    1. Copy a similar working core DataFrame method and update the required names.

```python
@pd.api.extensions.register_dataframe_accessor("ta")
class AnalysisIndicators(BasePandasObject):

    # ...
    # Trend
    # ...

    def ni(self, length=None, offset=None, **kwargs):
        close = self._get_column(kwargs.pop("close", "close"))
        result = ni(close=close, length=length, offset=offset, **kwargs)
        return self._post_process(result, **kwargs)
```
<br/>

That should be it!

Now you can run ```ni(close, length=10, **kwargs)``` either the _Standard_ way above or as a DataFrame Extension:

```python
import pandas as pd
import pandas_ta as ta

df = # ohlcv data
df.ta.ni(append=True)
```

Or via a _Custom Strategy_:
```python
import pandas as pd
import pandas_ta as ta

df = # ohlcv data
# (1) Create the Strategy
MyStrategy = ta.Strategy(
    name="NI Trend",
    ta=[
        {"kind": "ohlc4"},
        {"kind": "ni", "length": 12},
    ]
)

# (2) Run the Strategy
df.ta.strategy(MyStrategy, **kwargs)
```

### TLDR
Add and edit the file ```pandas_ta/trend/ni.py```. Then modify the three files: ```pandas_ta/trend/__init__.py```, ```pandas_ta/__init__.py```, and ```pandas_ta/core.py```

<br/>

Hope this helps! Please let me know (good/bad)! Future contributions are greatly appreciated! 😎 

Kind Regards,
KJ

**Comment by utpal4job** [comment]
hello i have tried to create custom indicator like you said but error appears. below are the error message:

File "/root/simulation_test/src/app.py", line 14, in <module>
    import pandas_ta as ta
  File "/root/myenv/lib/python3.11/site-packages/pandas_ta/__init__.py", line 1, in <module>
    from pandas_ta.core import *
  File "/root/myenv/lib/python3.11/site-packages/pandas_ta/core.py", line 14, in <module>
    from pandas_ta import Category, Imports, version
ImportError: cannot import name 'Category' from partially initialized module 'pandas_ta' (most likely due to a circular import) (/root/myenv/lib/python3.11/site-packages/pandas_ta/__init__.py)
